enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/RoleManager.java

```diff
@@ -42,4 +44,10 @@
      * @throws IOException
      */
     void removeUserFromRole( String username, String roleName ) throws IOException;
+
+    List<String> getAllRoleNames();
```

Comment:
I am not sure this is necessary. You shouldn't get duplicates unless you manually edit the role file and create them. And the result will be converted to a stream in the procedure, so the set type will not be preserved.